### PR TITLE
#78: Add Swagger 2 for API UI

### PR DESCRIPTION
**Pull request by petrokramar**
Implement Swagger 2 for API UI #78

### CODE REVIEW

File: src/main/java/juja/microservices/gamification/controller/AchievementController.java [hunk 1]
```diff
@@ -26,6 +30,15 @@
 
     @RequestMapping(value = "/achieve/daily", method = RequestMethod.POST)
     @ResponseBody
+    @ApiOperation(
+            value = "Add points for daily report",
+            notes = "This method adds points for daily report"
+    )
+    //TODO @ApiImplicitParams doesn`t works
+    //    @ApiImplicitParams(value - @ApiImplicitParam(name, value, dataType, required))
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/controller/AchievementController.java]: Please add values for these parameters. E.g, 
 @ApiImplicitParams(value = {@ApiImplicitParam( name = "email", value = "User's email", dataType = "application/json", required = true)})

File: src/main/java/juja/microservices/gamification/controller/AchievementController.java [hunk 2]
```diff
@@ -26,6 +30,15 @@
...
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "Returns achievement id.")
+    })
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/controller/AchievementController.java]: You can add in ApiResponses another ApiResponse with error codes and parameter 'response = ApiErrorMessage.class'

**Review by OlgaKulikova — COMMENTED** [review]
please look at the comments

**Comment by petrokramar** [comment]
I'm sorry. I've added into one pull request two tasks:
Implement Swagger 2 for API UI #78 and
Change responses in all achivements to List<String> ids #85.
It's done because task #78 depends on task #85 and code changes for the task #85 are minor.
I didn't use @ApiImplicitParams annotation in Swagger because default representation works well.

File: src/main/java/juja/microservices/gamification/controller/AchievementController.java [hunk 3]
```diff
@@ -27,14 +29,41 @@
...
     @ResponseBody
+    @ApiOperation(
+            value = "Add points for daily report",
+            notes = "This method adds points for daily report"
+    )
+    @ApiImplicitParams(value = {
+            @ApiImplicitParam(defaultValue = "d", example = "e", access = "a", allowableValues = "al", dataType = "{\n" +
+                    "    \"path\": \"/pets\",\n" +
+                    "    \"description\": \"Operations about pets.\"\n" +
+                    "}", name = "n", value = "zz")})
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/gamification/controller/AchievementController.java]: Я так понимаю, это было для проверки, как отображается. Думаю, уже не нужно?))

**Review by OlgaKulikova — APPROVED** [review]

**Pull request merged**